SteamKit2/SteamKit2/Steam/WebAPI/WebAPI.cs

```diff
@@ -222,6 +249,49 @@ public async Task<T> CallProtobufAsync<T>( HttpMethod method, string func, int v
                 return body;
             }
 
+            /// <summary>
+            /// Manually calls the specified Web API function with the provided details.
+            /// </summary>
+            /// <typeparam name="TRequest">Protobuf type of the request message.</typeparam>
+            /// <typeparam name="TResponse">Protobuf type of the response message.</typeparam>
+            /// <param name="func">The function name to call.</param>
+            /// <param name="version">The version of the function to call.</param>
+            /// <param name="request">A protobuf object representing arguments to be passed to the API.</param>
+            /// <param name="method">The http request method. Either "POST" or "GET".</param>
+            /// <returns>A <see cref="Task{T}"/> that contains object representing the results of the Web API call.</returns>
+            /// <exception cref="ArgumentNullException">The function name or request method provided were <c>null</c>.</exception>
+            /// <exception cref="HttpRequestException">An network error occurred when performing the request.</exception>
+            /// <exception cref="WebAPIRequestException">A network error occurred when performing the request.</exception>
+            /// <exception cref="ProtoException">An error occured when parsing the response from the WebAPI.</exception>
+            public async Task<WebAPIResponse<TResponse>> CallProtobufAsync<TResponse, TRequest>( HttpMethod method, string func, TRequest request, int version = 1 )
+                where TResponse : IExtensible, new()
+                where TRequest : IExtensible, new()
+            {
+                using var input_payload = new MemoryStream();
+                Serializer.Serialize<TRequest>( input_payload, request );
+
+                var base64 = Convert.ToBase64String( input_payload.ToArray() );
+                var args = new Dictionary<string, object?>
+                {
+                    { "input_protobuf_encoded", base64 },
+                };
+
+                var response = await CallAsyncInternal( method, func, version, args, "protobuf_raw" );
+                var eresult = EResult.Invalid;
+
+                using var stream = await response.Content.ReadAsStreamAsync().ConfigureAwait( false );
+
+                if ( response.Headers.TryGetValues( "x-eresult", out var result ) && int.TryParse( result.First(), out var resultInt ) )
```

Comment:
Should the `EResult` perhaps be nullable, i.e. null if it is not in the HttpResponseMessage, rather than defaulting to Invalid? Or does this cause significantly more typically-unneccesary work for consumers/

Reply:
Yeah I think nullable isn't particularly nice here. EResult.Invalid is 0.